### ISSUE TITLE
There should be a 'View mode' for Saved charts 

### ISSUE THREAD

**Issue by PriPatel** [lightdash/lightdash#1680]
### Design files
**Figma file and UI** 
https://www.figma.com/file/Mhuwki1Bj1akQIwPyr1gWw/Lightdash-product?node-id=2583%3A46348

**Prototype** 
https://www.figma.com/proto/Mhuwki1Bj1akQIwPyr1gWw/Lightdash-product?node-id=2820%3A48838&scaling=min-zoom&page-id=2583%3A46348

### The different modes
**When should the user land on `view mode`?** 
When they are opening a saved chart.

**When should the user land on `edit mode`?** 
When they hit 'edit chart' from the dashboard
When the user hits 'edit chart' in view mode.

**When should the user land on `explore mode`?** 
When they have opened a link to an unsaved chart
When they have clicked on `Explore from here`

### Summary of UI changes
- Sidebar animation of the drawer.
- Filter panel should be disabled.
- Show filters on hover.
- In `View mode`, you should only be able to `Export as` in the `charts` panel
- In `View mode`, you should only be able to `Download CSV` in the `Results` panel
- `Refresh dbt` button style is less prominent
- You should only be able to rename the chart/description in `Edit` mode ( this is different to how it works now) 
- Error messages should be shown on 'View mode' as well.
- You should **_not_** be able to `add to dashboard` / `delete` / `duplicate`/ `share` when in `Edit mode`. (hide the context menu and actions!

Tickets:

- [ ] #1709
- [ ] #2087 
- [ ] #2088 
- [ ] #2089 
- [x] #2090
- [ ] #2092
- [ ] #2076

**Comment by TuringLovesDeathMetal** [comment]
@PriPatel can this work be picked up? Or does this need to be broken into smaller tasks?

**Comment by PriPatel** [comment]
@TuringLovesDeathMetal Yup, I think it will need to be broken down further!

**Comment by PriPatel** [comment]
Also requires a kick-off session as I think its a big piece of work and Id like to understand if there are any holes in the logic!

**Comment by TuringLovesDeathMetal** [comment]
@PriPatel 

For these two questions, here are my thoughts:

> What happens with error messages? Do they still show in view mode?

Yes, I think they should still show in view mode. Was there a situation you ran into that made it feel like they shouldn't?

>What is the primary CTA?

For right now, "Explore from here"

Later, I imagine it will be drill-down options (#607) or there will be some way of sending users to `related charts` or `dashboards containing this chart`

**Comment by PriPatel** [comment]
@TuringLovesDeathMetal 
Thanks! Re. your question on error messages:
We thought about a case where you might save the link to 'Chart 1' somewhere, eg on a notion page.
The AE refreshes dbt, where they have deleted something that breaks 'Chart 1'.
What would the business user see when they next click on the link to the saved chart?


**Primary CTA**
This was in reference to some feedback about `Explore from here` being more prominent than 'Edit chart'. Do you agree with this?

**Comment by TuringLovesDeathMetal** [comment]
ooh, ya, I'd expect that the chart would break and they would see the error message. They could then send a screenshot of that message to an analytics engineer in their team and be like "hey, the thing you shared is broken".

Yep! `explore from here` is the more prominent action for sure :)

**Comment by PriPatel** [comment]
Thanks Katie !!

Ok - I've updated the original comment. Lets leave the CTAs as they are for now. We can change them in the future based on feedback, plus I don't think they are critical to the concept of view/edit/explore modes. 

 The only bit missing right now is a better 'share' button. Im working on this now but I don't think its a blocker! This can be a separate ticket but Ill let you know when it is done anyway!

**Comment by PriPatel** [comment]
An update : I decided that the user _shouldn't_ be able to 'add to dashboard', 'delete' or do any big actions when in `edit` mode. The actions button will be hidden when in edit mode. I have updated everything to reflect this - i hope this doesn't mess up any plans so far!

**This is for a few reasons:**
- One of them being to declutter the amount of options available to the user at any given time. In this case, let the options be to Save changes, or Cancel
- Another is to reduce confusion..eg they might try to add it to their to dashboard without saving changes and then be confused.
- Eventually I want the header to be really explicit in showing that they are editing. In some apps, the colour of the header is different and it literally says ' you are in edit mode'. I want to take baby steps to get there though :sweat_smile:
